### ISSUE TITLE
IDE0071 should suggest removing ToString on {ReadOnly}Span<char> for .NET 6+

### ISSUE THREAD

**Issue by Alex-Sob** [dotnet/roslyn#80132]
'[IDE0071](https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/style-rules/ide0071): Simplify interpolation' is not suggested for `Span<T>` or `ReadOnlySpan<T>` for .NET 6+.

String interpolation was [significantly improved](https://devblogs.microsoft.com/dotnet/string-interpolation-in-c-10-and-net-6) in .NET 6+ with the introduction of interpolated string handlers. For projects targeting .NET6+ code like this should trigger IDE0071:

```cs
var span = "abc".AsSpan();
var s = $"{span.ToString()}"; // Should trigger IDE0071
```

In older .NET versions including .NET Framework interpolated string expressions are lowered to `string.Format` calls, so IDE0071 should not be reported for projects targeting older .NET versions:

```cs
var span = "abc".AsSpan();
var s = $"{span.ToString()}"; // Should not trigger IDE0071, removing ToString will cause 'ReadOnlySpan<char> cannot be converted to object'
```

**Comment by Alex-Sob** [comment]
Prepared PR with the fix #80137
